src/Microsoft.AspNetCore.Razor.VSCode/src/Diagnostics/ReportIssueCreator.ts

```diff
@@ -0,0 +1,244 @@
+
+        return content;
+    }
+
+    // Protected for testing
+    protected async getProjectedCSharp(razorDocument: IRazorDocument) {
+        let csharpContent = `////////////////////// Projected CSharp as seen by extension ///////////////////////
+${razorDocument.csharpDocument.getContent()}
+
+
+////////////////////// Projected CSharp as seen by VSCode ///////////////////////`;
+
+        const csharpTextDocument = await this.vscodeApi.workspace.openTextDocument(razorDocument.csharpDocument.uri);
+        if (csharpTextDocument) {
+            csharpContent = `${csharpContent}
+${csharpTextDocument.getText()}`;
+        } else {
+            csharpContent = `${csharpContent}
+Unable to resolve VSCode's version of CSharp`;
+        }
+
+        return csharpContent;
+    }
+
+    // Protected for testing
+    protected async getProjectedHtml(razorDocument: IRazorDocument) {
+        let htmlContent = `////////////////////// Projected Html as seen by extension ///////////////////////
+${razorDocument.htmlDocument.getContent()}
+
+
+////////////////////// Projected Html as seen by VSCode ///////////////////////`;
+
+        const htmlTextDocument = await this.vscodeApi.workspace.openTextDocument(razorDocument.htmlDocument.uri);
+        if (htmlTextDocument) {
+            htmlContent = `${htmlContent}
+${htmlTextDocument.getText()}`;
+        } else {
+            htmlContent = `${htmlContent}
+Unable to resolve VSCode's version of Html`;
+        }
+
+        return htmlContent;
+    }
+
+    // Protected for testing
+    protected getExtensionVersion(): string {
+        const extension = this.vscodeApi.extensions.getExtension(razorExtensionId);
+        if (!extension) {
+            return 'Unable to find Razor extension version.';
+        }
+        return extension.packageJSON.version;
+    }
+
+    // Protected for testing
+    protected getInstalledExtensions() {
+        const extensions: Array<vscode.Extension<any>> = this.vscodeApi.extensions.all
+            .filter(extension => extension.packageJSON.isBuiltin === false);
+
+        return extensions.sort(this.sortExtensions);
+    }
+
+    // Protected for testing
+    protected sortExtensions(a: vscode.Extension<any>, b: vscode.Extension<any>): number {
+        const aName = a.packageJSON.name.toLowerCase();
+        const bName = b.packageJSON.name.toLowerCase();
+        if (aName < bName) {
+            return -1;
+        }
+        if (aName > bName) {
+            return 1;
+        }
+        return 0;
+    }
+
+    // Protected for testing
+    protected generateExtensionTable() {
+        const extensions = this.getInstalledExtensions();
+        if (extensions.length <= 0) {
+            return 'none';
+        }
+
+        const tableHeader = `|Extension|Author|Version|\n|---|---|---|`;
+        const table = extensions.map(
+            (e) => `|${e.packageJSON.name}|${e.packageJSON.publisher}|${e.packageJSON.version}|`).join(os.EOL);
```

Comment:
We're using `\n` elsewhere, so it's odd to use `os.EOL` here.

Reply:
Good catch. A lot of this code was pulled from OmniSharp and I replaced small pieces of it. This was an oversight which was half-replaced.